### ISSUE TITLE
Usage of TranslationWriter & LocalStorage

### ISSUE THREAD

**Issue by odolbeau** [php-translation/symfony-bundle#392]
Discussing about the configuration of the local storage with @Guite (see https://github.com/php-translation/symfony-storage/issues/48) I found something which looks strange to me.

It looks like we don't use the same logic between `download` & `sync` command (see comment https://github.com/php-translation/symfony-storage/issues/48#issuecomment-575671903).

Shouldn't we ensure we always use the local storage everywhere & never rely on the TranslationWriter directly?
This could allow us to simplify the configuration needed to change the output format (which need to be specified twice right now).

**Comment by nicwortel** [comment]
Does this also include the `extract` command? I ran into an issue recently where the order of translations is completely different after running the `translation:extract` command as compared to running `translation:download`. I'm assuming because they use different logic to dump the translation file, although it could also be simply because the translations are not ordered at all.

This is of course very annoying if you commit the translation file and everything is reordered after you run `translation:extract`...

Not sure if I should create a seperate issue for it?

**Comment by odolbeau** [comment]
Yes, the `extract` cmmand is also included & both issues are probably related.
I'll try to create a PR to address this problem this week or the next time if I can! :)

**Comment by nicwortel** [comment]
@odolbeau were you able to make any progress on this issue? Is there anything I can do to help, or do you have any pointers for me if I were to try to fix this issue myself?

#398 is really affecting me and my team (@leviy) as we have to manually track down all extracted translations after running `translation:extract`, copy them to our clipboard, revert the changes made to `messages.en.xlf` and then add the new translations to the bottom of the file... so it would be great if we can resolve this and I would be happy to help :smiley:

**Comment by odolbeau** [comment]
Hi!
We discuss this issue and a few others with @welcoMattic 
We would like to refactor a bit this bundle following those steps:
* Merge interfaces in `php-translation/common` (https://github.com/php-translation/common/issues/30)
* Create a `ChainStorage` in `php-translation/common` (https://github.com/php-translation/common/pull/29)
* Remove the `StorageService` from the bundle & deal with the `ChainStorage` directly in concerned commands.

This should simplify the bundle & harmonize the behavior of all commands.
